diagnostics/collector.go

```diff
@@ -165,16 +184,53 @@ func (c *DBCollector) writeRawQueryContent(ctx context.Context, writer *zip.Writ
 	return err
 }
 
-func (c *DBCollector) writeTableContent(ctx context.Context, writer *zip.Writer, name string) error {
-	q := sq.Select("*").From(name)
-	return c.writeQueryContent(ctx, writer, name, q)
-}
-
-func (c *DBCollector) writeTableCount(ctx context.Context, writer *zip.Writer, name string) error {
+func (c *Collector) writeTableCount(ctx context.Context, writer *zip.Writer, name string) error {
 	q := sq.Select("COUNT(*)").From(name)
 	return c.writeQueryContent(ctx, writer, name+"_count", q)
 }
 
+func (c *Collector) rangesStats(ctx context.Context, writer *zip.Writer) []error {
+	var errs []error
+	repos, _, err := c.cataloger.ListRepositories(ctx, -1, "")
+	if err != nil {
+		// Cannot list repos, nothing to do..
+		errs = append(errs, fmt.Errorf("listing repositories: %w", err))
+		return errs
+	}
+
+	rangesFile, err := writer.Create("graveler-ranges")
+	if err != nil {
+		errs = append(errs, fmt.Errorf("creating ranges file: %w", err))
+		return errs
+	}
+
+	csvWriter := csv.NewWriter(rangesFile)
+	defer csvWriter.Flush()
+	if err := csvWriter.Write([]string{"repo", "type", "count(max:1000)"}); err != nil {
+		errs = append(errs, fmt.Errorf("writing headers: %w", err))
+	}
+
+	for _, repo := range repos {
+		count := 0
+		counter := func(id string) error {
+			count++
+			return nil
+		}
+
+		if err := c.adapter.Walk(block.WalkOpts{
+			StorageNamespace: repo.StorageNamespace,
+			Prefix:           c.pyramidCfg.BlockStoragePrefix,
+		}, counter); err != nil {
+			errs = append(errs, fmt.Errorf("listing ranges and meta-ranges: %w", err))
+		}
+		if err := csvWriter.Write([]string{repo.Name, "range", strconv.Itoa(count)}); err != nil {
```

Comment:
You still try to write the count to the CSV even if the previous step failed, it could be misleading.

Reply:
I agree but think it's valuable. 
Failing on the first iteration isn't like failing on the n'th so writing the number of ranges seen is good debugging information.